sdk/core/src/options.rs

```diff
@@ -0,0 +1,153 @@
+use crate::policies::{ExponentialRetryPolicy, FixedRetryPolicy, NoRetryPolicy, Policy};
+use crate::{new_http_client, HttpClient};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Client options allow customization of policies, retry options, and more.
+#[derive(Clone, Debug, Default)]
+pub struct ClientOptions {
+    // TODO: Expose transport override.
+    /// Policies called per call.
+    pub(crate) per_call_policies: Vec<Arc<dyn Policy>>,
+
+    /// Policies called per retry.
+    pub(crate) per_retry_policies: Vec<Arc<dyn Policy>>,
+
+    /// Retry options.
+    pub(crate) retry: RetryOptions,
+
+    /// Telemetry options.
+    pub(crate) telemetry: TelemetryOptions,
+
+    /// Transport options.
+    pub(crate) transport: TransportOptions,
+}
+
+impl ClientOptions {
+    setters! {
+        per_call_policies: Vec<Arc<dyn Policy>> => per_call_policies,
+        per_retry_policies: Vec<Arc<dyn Policy>> => per_retry_policies,
```

Comment:
We may want to expose a setter for these that returns an `&mut Vec<Arc<dyn Policy>>` since it's probably more likely that users want to add policies rather than outright replace the entire collection.

Reply:
Do we need to keep these with the new `mut` setters?